src/event.js

```diff
@@ -36,24 +11,16 @@ const emitEvent = (kinesis, event, config) => {
   return kinesis.putRecord(params).promise();
 };
 
-const emitEvents = (kinesis, events, config) => {
-  const records = events.map(event => {
-    const enrichedEvent = enrichMeta(event, config.appName, config.ipv4);
-    return {
-      Data: JSON.stringify(enrichedEvent),
-      PartitionKey: config.partitionKey || partitionKey(enrichedEvent),
-    }
-  })
-
-  const params = {
-    Records: records,
-    StreamName: config.kinesisStream.resource
-  };
-
-  return kinesis.putRecords(params).promise();
+const emitEventWithRetry = async (kinesis, event, config, retries) => {
+  try {
+      return await emitEvent(kinesis, event, config)
+  } catch(error) {
+      if (retries === 0) throw error;
+      return await emitEventWithRetry(kinesis, event, config, retries - 1);
+  }
 };
 
-module.exports = {
-  emitEvent,
-  emitEvents,
+module.exports = (kinesis, event, config) => {
```

Comment:
```suggestion
module.exports = async (kinesis, event, config) => {
```

Seems to work without, but I think with it it should also print warnings if someone tries to use it without awaiting, which it might not do without the `async`